src/modules/deliveryoptimization/src/lib/DeliveryOptimization.c

```diff
@@ -0,0 +1,417 @@
+        }
+
+        if (NULL != rootValue)
+        {
+            json_value_free(rootValue);
+        }
+
+        // Reset status and payload if JSON file could not be parsed or property not found, as it may yet have to be configured.
+        if (MMI_OK != status)
+        {
+            const char* emptyJsonPayload = NULL;
+            if (JSONNumber == jsonPropertyType)
+            {
+                emptyJsonPayload = "0";
+            }
+            else if (JSONString == jsonPropertyType)
+            {
+                emptyJsonPayload = "\"\"";
+            }
+
+            *payloadSizeBytes = strlen(emptyJsonPayload);
+            *payload = (MMI_JSON_STRING)malloc(*payloadSizeBytes);
+
+            if (NULL != payload)
+            {
+                strncpy(*payload, emptyJsonPayload, *payloadSizeBytes);
+                status = MMI_OK;
+            }
+            else 
+            {
+                OsConfigLogError(DeliveryOptimizationGetLog(), "MmiGet failed to allocate %d bytes", *payloadSizeBytes + 1);
+                *payloadSizeBytes = 0;
+                status = ENOMEM;
+            }
+        }
+    }
+
+    if (IsFullLoggingEnabled())
+    {
+        OsConfigLogInfo(DeliveryOptimizationGetLog(), "MmiGet(%p, %s, %s, %.*s, %d) returning %d", clientSession, componentName, objectName, *payloadSizeBytes, *payload, *payloadSizeBytes, status);
+    }
+
+    return status;
+}
+
+int DeliveryOptimizationMmiSet(MMI_HANDLE clientSession, const char* componentName, const char* objectName, const MMI_JSON_STRING payload, const int payloadSizeBytes)
+{
+    int status = MMI_OK;
+
+    if ((NULL == componentName) || (NULL == objectName) || (NULL == payload) || (payloadSizeBytes < 0))
+    {
+        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiSet(%s, %s, %p, %d) called with invalid arguments", componentName, objectName, payload, payloadSizeBytes);
+        status = EINVAL;
+        return status;
+    }
+
+    if (!IsValidSession(clientSession))
+    {
+        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiSet(%s, %s) called outside of a valid session", componentName, objectName);
+        status = EINVAL;
+    }
+        
+    if ((MMI_OK == status) && (strcmp(componentName, g_deliveryOptimizationComponentName)))
+    {
+        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiSet called for an unsupported component name (%s)", componentName);
+        status = EINVAL;
+    }
+        
+    if ((MMI_OK == status) && (strcmp(objectName, g_desiredDeliveryOptimizationPoliciesObjectName)))
+    {
+        OsConfigLogError(DeliveryOptimizationGetLog(), "MmiSet called for an unsupported object name (%s)", objectName);
+        status = EINVAL;
+    }
+
+    if (MMI_OK == status)
+    {
+        // Make sure that payload is null-terminated for json_parse_string.
+        char* buffer = malloc(payloadSizeBytes + 1);
+        if (NULL != buffer)
+        {
+            buffer[payloadSizeBytes] = '\0';

```

Comment:
>buffer[payloadSizeBytes] = '\0'; [](http://example.com/codeflow?start=12&length=32)

Init as ={0} instead, same as it's done everywhere else in this project.

Reply:
Removed this line because no longer needed when memset all to zero instead. As per previous comment.